tests/e2e/basic-ingest-redis-serving.py

```diff
@@ -63,101 +63,164 @@ def client(core_url, serving_url, allow_dirty):
-        {
-            "datetime": [datetime.utcnow().replace(tzinfo=pytz.utc) for _ in
-                         range(5)],
-            "customer_id": [offset + inc for inc in range(5)],
-            "daily_transactions": [np.random.rand() for _ in range(5)],
-            "total_transactions": [512 for _ in range(5)],
-        }
-    )
-
+    df_dict = {
+        "datetime": [ingest_time.replace(tzinfo=pytz.utc) for _ in
+                     range(n_size)],
+    }
+    for entity_name in entities:
+        df_dict[entity_name] = list(range(1, n_size + 1))
+    for feature_name in features:
+        df_dict[feature_name] = [np.random.rand() for _ in range(n_size)]
+    return pd.DataFrame(df_dict)
+
+
+@pytest.fixture(scope="module")
+def ingest_time():
+    return datetime.utcnow()
+
+@pytest.fixture(scope="module")
+def cust_trans_df(ingest_time):
+    return basic_dataframe(entities=["customer_id"],
+                           features=["daily_transactions", "total_transactions"],
+                           ingest_time=ingest_time,
+                           n_size=5)
+
+@pytest.fixture(scope="module")
+def driver_df(ingest_time):
+    return basic_dataframe(entities=["driver_id"],
+                           features=["rating", "cost"],
+                           ingest_time=ingest_time,
+                           n_size=5)
 
 @pytest.mark.timeout(45)
 @pytest.mark.run(order=10)
 def test_basic_register_feature_set_success(client):
-    # Load feature set from file
+    # Register feature set without project
     cust_trans_fs_expected = FeatureSet.from_yaml("basic/cust_trans_fs.yaml")
+    driver_fs_expected = FeatureSet.from_yaml("basic/driver_fs.yaml")
+    client.apply(cust_trans_fs_expected)
+    client.apply(driver_fs_expected)
+    cust_trans_fs_actual = client.get_feature_set("customer_transactions")
+    assert cust_trans_fs_actual == cust_trans_fs_expected
+    driver_fs_actual = client.get_feature_set("driver")
+    assert driver_fs_actual == driver_fs_expected
 
+    # Register feature set with project
+    cust_trans_fs_expected = FeatureSet.from_yaml("basic/cust_trans_fs.yaml")
     client.set_project(PROJECT_NAME)
-
-    # Register feature set
     client.apply(cust_trans_fs_expected)
-
-    cust_trans_fs_actual = client.get_feature_set(name="customer_transactions")
-
+    cust_trans_fs_actual = client.get_feature_set("customer_transactions",
+                                                  project=PROJECT_NAME)
     assert cust_trans_fs_actual == cust_trans_fs_expected
 
-    if cust_trans_fs_actual is None:
-        raise Exception(
-            "Client cannot retrieve 'customer_transactions' FeatureSet "
-            "after registration. Either Feast Core does not save the "
-            "FeatureSet correctly or the client needs to wait longer for FeatureSet "
-            "to be committed."
-        )
-
+    # reset client's project to default for other tests
+    client.set_project(DEFAULT_PROJECT)
```

Comment:
I want to make sure that users dont have to set the project. Can we perhaps recreate the client or force clear this project?

Reply:
Changed this to `client.set_project()` to reset to default project.